criu/mount.c

```diff
@@ -491,15 +491,15 @@ static int try_resolve_ext_mount(struct mount_info *info)
 			int len;
 
 			len = strlen(val) + sizeof("dev[]");
-			source = xmalloc(len);
-			if (source == NULL)
+			source = xrealloc(info->source, len);
+			if (source == NULL) {
+				xfree(info->source);
```

Comment:
It is dangerous. mnt_entry_free() will call xfree(info->source) again and it will be a double free fault. You don't need to call xfree here or you need to set info->source to NULL after that. I would prefer to not call xfree.